sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/PubsubIO.java

```diff
@@ -1390,6 +1392,7 @@ static <T> Builder<T> newBuilder(
       builder.setFormatFn(formatFn);
       builder.setBadRecordRouter(BadRecordRouter.THROWING_ROUTER);
       builder.setBadRecordErrorHandler(new DefaultErrorHandler<>());
+      builder.setValidate(true);
```

Comment:
This is the entry method for all writes, so it's enough to `setValidate(true)` here and remove it from the other methods.

It's better to only set it here, otherwise validation will be set to true even when someone doesn't want it. Take the following for example:

```java
PubsubIO
  .writeStrings() // validate = true
  .withoutValidation() // validate = false
  .to("topic"); // validate = true
 ```

Reply:
I agree. 
[24859f8](https://github.com/apache/beam/pull/32465/commits/24859f869d28d9ae3efbcbf803c249981725725a)